test/suggestion.cpp

```diff
@@ -495,4 +511,32 @@ namespace {
       );
     }
   }
+
+  TEST(Suggestion, titleSnippet) {
+    TempZimArchive tza("testzim");
+
+    const zim::Archive archive = tza.createZimFromTitles({
+      "this is a straight run of matching words",
+      "this is a broken set of likely words",
+      "this is a a long title to ensure that the snippets generated contain the entire title even if match is one word"
+    });
+
+    EXPECT_SNIPPET_EQ(
+      archive,
+      "straight run of matching",
+      "this is a <b>straight</b> <b>run</b> <b>of</b> <b>matching</b> words"
+    );
+
+    EXPECT_SNIPPET_EQ(
+      archive,
+      "broken likely",
+      "this is a <b>broken</b> set of <b>likely</b> words"
+    );
+
+    EXPECT_SNIPPET_EQ(
+      archive,
+      "generated",
+      "this is a a long title to ensure that the snippets <b>generated</b> contain the entire title even if match is one word"
+    );
```

Comment:
It could also be interesting to check snippet with query "this is". To check the behavior when there are several results.

Reply:
Adding a new test case for this.